### PR TITLE
1.2.0 - Cleanup stabilized extension ID

### PR DESCRIPTION
**Pull request by abique**
This PR changes the stabilized extension id and adds the compatibility extension ID to mitigated the break.

### CODE REVIEW

**Comment by baconpaul** [comment]
This looks great to me.

We will do an update to clap-helpers once 1.2.0 is stabilized also? If so then i can pull all to head in clap juce extensions and clap info also.

**Comment by abique** [comment]
> This looks great to me.
> 
> We will do an update to clap-helpers once 1.2.0 is stabilized also? If so then i can pull all to head in clap juce extensions and clap info also.

Yes we need to merge it first :-)

**Comment by baconpaul** [comment]
ha yeah! i'm generally around next 2 weeks so given 48 hours notice or so and i can get helpers and extensions and surge staged.

File: ChangeLog.md [hunk 1]
```diff
@@ -13,8 +17,17 @@
...
+
+We changed the extension ID in the process of stabilization which leads to a **break**.
+
+To mitigate this transition, we provide a compatibily extension ID which can be used to match and use the draft extension as they are 100% compatible.
+
+For example, `CLAP_EXT_CONTEXT_MENU` for the stable ID and `CLAP_EXT_CONTEXT_MENU_COMPAT` for the draft ID.
+
+As you can see in [extension-id](conventions/extension-id.md), we introduce some rules, so this kind of breaks won't happen again.
+
+We may decide to remove `CLAP_EXT_CONTEXT_MENU_COMPAT` in the future once their usage becomes anecdotic.
```
Comment (messmerd) [inline ChangeLog.md:30]:
```suggestion
To mitigate this transition, we've provided compatibility extension IDs which can be used to match and use the latest draft extensions as they are 100% compatible.

For example, `CLAP_EXT_CONTEXT_MENU` for the stable ID and `CLAP_EXT_CONTEXT_MENU_COMPAT` for the draft ID.

As you can see in [extension-id](conventions/extension-id.md), we introduced some rules, so this kind of break won't happen again.

We may decide to remove the `*_COMPAT` IDs in the future once their usage becomes antiquated.
```

File: conventions/extension-id.md [hunk 2]
```diff
@@ -28,5 +28,6 @@ Everything about the extension id symmetrically applies to factory id.
 ## History
 
 Before this document was written, existing extensions didn't honor these rules.
-Some stable extensions include the string `draft` in their ID.
-We decided to keep those in order to maintain binary compatibility.
\ No newline at end of file
+We wanted to stabilize some extension without breaking the compatibility, yet the extension ID contained the string `draft`.
+While this isn't user facing, we wanted to get rid of it, so we updated the extension ID according to this document and
+introduced a `XXX_COMPAT` ID to provide backward compatibility with the draft version.
```
Comment (messmerd) [inline conventions/extension-id.md:33]:
```suggestion
We wanted to stabilize some draft extensions without breaking compatibility, yet their extension IDs contained the string `draft`.
While these strings weren't user-facing, we still wanted to remove them, so we updated the extension IDs according to this document and introduced IDs with `_COMPAT` suffixes to provide backward compatibility with the draft versions.
```
I tweaked the wording a bit.

Also, I think the "History" section should mention which version introduced these conventions.
Maybe line 30 could be changed like this?
```diff
-Before this document was written, existing extensions didn't honor these rules.
+Before version 1.2.0 when this document was written, existing extensions didn't honor these rules.
```

File: include/clap/ext/remote-controls.h [hunk 3]
```diff
@@ -31,9 +31,10 @@
 // Pressing that button once gets you to the first page of the section.
 // Press it again to cycle through the section's pages.
 
-// This extension ID contains `draft', yet it is stable.
-// See conventions/extension-id.md for more info.
-static CLAP_CONSTEXPR const char CLAP_EXT_REMOTE_CONTROLS[] = "clap.remote-controls.draft/2";
+static CLAP_CONSTEXPR const char CLAP_EXT_REMOTE_CONTROLS[] = "clap.remote-controls/2";
+
+// The latest draft is 100% compatible
```
Comment (messmerd) [inline include/clap/ext/remote-controls.h:36]:
```suggestion
// The latest draft is 100% compatible
// This compat ID may be removed in 2026.
```
Did you mean to leave this out?

**Review by messmerd — COMMENTED** [review]
Some typo fixes and wording suggestions

File: ChangeLog.md (hunk 1 above)
Comment (abique) [reply ChangeLog.md:30]: Thanks 👍

File: include/clap/ext/remote-controls.h (hunk 3 above)
Comment (abique) [reply include/clap/ext/remote-controls.h:36]: I've overlooked it.

**Pull request merged**